rosidl_typesupport_introspection_cpp/resource/srv__type_support.cpp.em

```diff
@@ -110,7 +113,7 @@ extern "C"
 ROSIDL_TYPESUPPORT_INTROSPECTION_CPP_PUBLIC
 const rosidl_service_type_support_t *
 ROSIDL_TYPESUPPORT_INTERFACE__SERVICE_SYMBOL_NAME(rosidl_typesupport_introspection_cpp, @(spec.pkg_name), @(spec.srv_name))() {
```

Comment:
I think the macro `ROSIDL_TYPESUPPORT_INTERFACE__SERVICE_SYMBOL_NAME` needs a parameter that is either `srv` or `action` just like `ROSIDL_TYPESUPPORT_INTERFACE__MESSAGE_SYMBOL_NAME` has. Otherwise this creates a symbol with `srv` in the name that conflicts when the same service name is in an action and a `srv` folder.

There's a branch with a duplicate service and message for testing here https://github.com/ros2/rcl_interfaces/tree/sloretz/msg_srv_in_action/test_msgs

Here's the compiler error when trying to build it (when skipping rosidl_generator_py)

```
CMakeFiles/test_msgs__rosidl_typesupport_fastrtps_cpp.dir/rosidl_typesupport_fastrtps_cpp/test_msgs/action/dds_fastrtps/primitives__type_support.cpp.o: In function `rosidl_typesupport_fastrtps_cpp__get_service_type_support_handle__test_msgs__srv__Primitives':
/home/developer/workspaces/ros2-dev0/build/test_msgs/rosidl_typesupport_fastrtps_cpp/test_msgs/action/dds_fastrtps/primitives__type_support.cpp:62: multiple definition of `rosidl_typesupport_fastrtps_cpp__get_service_type_support_handle__test_msgs__srv__Primitives'
CMakeFiles/test_msgs__rosidl_typesupport_fastrtps_cpp.dir/rosidl_typesupport_fastrtps_cpp/test_msgs/srv/dds_fastrtps/primitives__type_support.cpp.o:/home/developer/workspaces/ros2-dev0/build/test_msgs/rosidl_typesupport_fastrtps_cpp/test_msgs/srv/dds_fastrtps/primitives__type_support.cpp:62: first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [libtest_msgs__rosidl_typesupport_fastrtps_cpp.so] Error 1
make[1]: *** [CMakeFiles/test_msgs__rosidl_typesupport_fastrtps_cpp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
CMakeFiles/test_msgs__rosidl_typesupport_introspection_cpp.dir/rosidl_typesupport_introspection_cpp/test_msgs/action/primitives__type_support.cpp.o: In function `rosidl_typesupport_introspection_cpp__get_service_type_support_handle__test_msgs__srv__Primitives':
/home/developer/workspaces/ros2-dev0/build/test_msgs/rosidl_typesupport_introspection_cpp/test_msgs/action/primitives__type_support.cpp:103: multiple definition of `rosidl_typesupport_introspection_cpp__get_service_type_support_handle__test_msgs__srv__Primitives'
CMakeFiles/test_msgs__rosidl_typesupport_introspection_cpp.dir/rosidl_typesupport_introspection_cpp/test_msgs/srv/primitives__type_support.cpp.o:/home/developer/workspaces/ros2-dev0/build/test_msgs/rosidl_typesupport_introspection_cpp/test_msgs/srv/primitives__type_support.cpp:103: first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [libtest_msgs__rosidl_typesupport_introspection_cpp.so] Error 1
make[1]: *** [CMakeFiles/test_msgs__rosidl_typesupport_introspection_cpp.dir/all] Error 2
make: *** [all] Error 2
```